pkg/k8s/client.go

```diff
@@ -35,82 +37,94 @@ const (
 
 // KubeInfo holds the data required to talk to a cluster
 type KubeInfo struct {
-	Kubeconfig   string
-	context      string
-	namespace    string
-	clientConfig clientcmd.ClientConfig
+	flags       *pflag.FlagSet
+	configFlags *genericclioptions.ConfigFlags
+	config      *rest.Config
+	namespace   string
 }
 
 // NewKubeInfo returns a useable KubeInfo, handling optional
 // kubeconfig, context, and namespace.
 func NewKubeInfo(configfile, context, namespace string) *KubeInfo {
-	// Find the correct kube config file
-	configfilesearch := clientcmd.NewDefaultClientConfigLoadingRules()
-	if len(configfile) != 0 {
-		configfilesearch.ExplicitPath = configfile
-	}
+	flags := pflag.NewFlagSet("KubeInfo", pflag.PanicOnError)
```

Comment:
I wrote 15 days ago:
> ~~I'm not entirely comfortable with `pflag.PanicOnError`.  `pflag.ContinueOnError` will have `flags.Parse()` return an error, which we should return to the caller.~~
>
> Can we add a comment that since `args` is constructed by us, and there should never be an error, it's OK to say `PanicOnError` instead of `ContinueOnError`?

Reply:
Done